docs/assignment2.md

```diff
@@ -1 +1,157 @@
+
+#### Initial Setup Sequence Diagram
+
+ ![Initial Setup Sequence Diagram](https://github.com/sarahpazik/software-design-vu/blob/Assignment2/Initial%20Setup%20Sequence%20Diagram.png)
+
+This diagram describes the interaction a user undertakes when selecting the story file to use and how the program initializes all of the necessary objects from the json data. This step occurs after the program has initially been run and the user is prompted to input the name of the story file they intend to use. First, the user will type the absolute file path to this json. Here, the main method of the **main** class will begin to attempt parsing. It will check if the file exists, can be open, and can be read. If there is a file error, an exception will be thrown, the user will be informed that they supplied an invalid file, and program execution will terminate. Otherwise, as can be seen in the diagram, the user enters a valid file and parsing begins. A correctly formatted file will describe the start **room** and the end **room** by their names. It will also have a full description of each **room** in the game, including its name, a list of **room** objects it connects to, a list of **item** objects inside of it, and a string script that is associated with the **room** object’s role in the story. **Item** objects will have a name field and a usage description field. The main method will process and save the start **room** name, end **room** name, and list of **room** descriptions. Next, it will traverse this list of **room** objects as described by the json in a loop. Inside of this loop, it will traverse the list of **item** objects as described by the json in the **room**. Here, all of the **item** objectss in a room will be initialized, passing their name and usage data from the json to the **item** constructor. They are stored in a map from their name to the **item** object. After all the **item** objects in the **room** are traversed and mapped, the process of initializing the **room** object begins. All of the data from the json about the **room** object’s name, connections, and script is parsed and, in conjunction with the previously created map of **item** objects, passed to the **room** constructor. Now that the base world is built, the user is prompted for his or her name.
+
```

Comment:
-Description: Great
-Model: Great, your loop fragments should specify min, max number of iterations. Also, the guard condition not clear enough.